polyfill/intersection-observer.js

```diff
@@ -971,6 +971,10 @@ function getParentNode(node) {
   }
 
   if (parent && parent.assignedSlot) {
+    // If the parent is assigned into shadow root slot
+    if (parent.assignedSlot.parentNode.host) {
```

Comment:
It might be better to move this part before the previous `if (parent.nodeType == 11 && parent.host)` and do something like:

```
if (parent && parent.assignedSlot) {
  parent = parent.assignedSlot.parentNode;
}
```

Then the `if (parent.nodeType == 11 && parent.host)` will do the right thing.